avocado/utils/software_manager/main.py

```diff
@@ -32,18 +34,21 @@ def main():
             log.info("Packages %s installed successfully", args)
         else:
             log.error("Failed to install %s", args)
+            exitcode |= exit_codes.UTILITY_MEDIUM
 
     elif action == 'remove':
         if software_manager.remove(args):
             log.info("Packages %s removed successfully", args)
         else:
             log.error("Failed to remove %s", args)
+            exitcode |= exit_codes.UTILITY_MEDIUM
```

Comment:
AFAICT only one action will be invoked at a time, right? So there's no need to "OR" the exit codes.  Also, I don't think we need to flag the "level of success", let's keep it as a simple "success == 0", "failure > 0" (but just one would be enough for now) and "crash == -1".

Reply:
> AFAICT only one action will be invoked at a time, right? So there's no need to "OR" the exit codes. 

Good catch, I'll change that.

> Also, I don't think we need to flag the "level of success", let's keep it as a simple "success == 0", "failure > 0" (but just one would be enough for now) and "crash == -1".

The intention here was to differentiate a failed check from a failed action. I understand a 'False' return during a check as a  "light" fail. When doing a package install, for example, if it fails, I would classify it at a higher level. On the other hand, it would not be a problem using just one exit code for a failed action as other programs would use it, and a check and an action would be executed in different flows. My only concern is having the same signal with likely different meanings, depending on what we are executing, but this is a personal preference.

Reply:
> > AFAICT only one action will be invoked at a time, right? So there's no need to "OR" the exit codes.
> 
> Good catch, I'll change that.
> 
> > Also, I don't think we need to flag the "level of success", let's keep it as a simple "success == 0", "failure > 0" (but just one would be enough for now) and "crash == -1".
> 
> The intention here was to differentiate a failed check from a failed action. I understand a 'False' return during a check as a "light" fail. When doing a package install, for example, if it fails, I would classify it at a higher level. On the other hand, it would not be a problem using just one exit code for a failed action as other programs would use it, and a check and an action would be executed in different flows. My only concern is having the same signal with likely different meanings, depending on what we are executing, but this is a personal preference.

About distinguishing between a failed check and a failed action, isn't just one (either check or action) possible in each invocation? If so, it's possible to infer what happened from the invocation.  What I mean is:

1) avocado-software-manager check-installed bash
1.1) exit 0 means successful check
1.2) exit > 0 means failed check
1.3) exit == -1 means crash

2) `avocado-software-manager install bash`
2.1) exit 0 means successful action
2.2) exit > 0 means failed action
2.3) exit == -1 means crash

I think the caller should be responsible for determining the seriousness of the failure.  The goal of the (exit code) interface is to provide unambiguous information, and I think this simpler version achieves it.

Reply:
Yep, that is what I meant with the statement starting at "On the other hand..." in my previous comment, except by your final conclusion. I'll send a v2 and we can continue the discussion on having a new exit code file vs sysexits.

Reply:
@willianrampazzo please see my previous comment here.